src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/statusindicator/StatusIndicatorSkin.kt

```diff
@@ -0,0 +1,230 @@
+package org.wycliffeassociates.otter.jvm.statusindicator.control
+
+import javafx.beans.property.DoubleProperty
+import javafx.beans.property.ObjectProperty
+import javafx.beans.property.SimpleDoubleProperty
+import javafx.beans.property.SimpleObjectProperty
+import javafx.geometry.HPos
+import javafx.geometry.Insets
+import javafx.geometry.VPos
+import javafx.scene.control.SkinBase
+import javafx.scene.layout.Background
+import javafx.scene.layout.BackgroundFill
+import javafx.scene.layout.CornerRadii
+import javafx.scene.layout.StackPane
+import javafx.scene.paint.Color
+import javafx.scene.paint.CycleMethod
+import javafx.scene.paint.LinearGradient
+import javafx.scene.paint.Stop
+import tornadofx.*
+
+
+class StatusIndicatorSkin(control: StatusIndicator) : SkinBase<StatusIndicator>(control) {
+
+    private var bar: StackPane
+    private var track: StackPane
+    private var barWidth: Double = 0.0
+
+
+    private var invalidBar = true
+    private val localProgressProperty: DoubleProperty = SimpleDoubleProperty(0.0)
+    private val skinPrimaryFill: ObjectProperty<Color> = SimpleObjectProperty<Color>(Color.RED)
+    private val skinAccentFill: ObjectProperty<Color> = SimpleObjectProperty<Color>(Color.RED)
+    private val skinTrackColor: ObjectProperty<Color> = SimpleObjectProperty<Color>(Color.WHITE)
+    private val barHeight: DoubleProperty = SimpleDoubleProperty(0.0)
+    private val indicatorRadius: DoubleProperty = SimpleDoubleProperty(0.0)
+
+
+    init {
+        //need to define super(statusindicator)? not sure if this is required atm
+        bar = StackPane()
+        bar.styleClass.setAll("bar")
+
+        track = StackPane()
+        track.styleClass.setAll("track")
+
+        children.setAll(track, bar)
+
+        control.widthProperty().onChange { observable -> invalidBar = true }
+        control.heightProperty().onChange { observable -> invalidBar = true }
+        control.primaryFillProperty.onChange { observable -> updateBarFill(control.primaryFill, control.accentFill) }
+        control.accentFillProperty.onChange { observable -> updateBarFill(control.primaryFill, control.accentFill) }
+        control.progressProperty.onChange { progress ->
```

Comment:
I don't have this open in an editor, so I can't test, but should the `observable ->` and `progress ->` stuff above be omitted?